app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,55 @@
+class AlbumsController < ApplicationController
+
+  def index
+    @albums = Album.all.order(:upvotes).reverse
+  end
+
+  def show
+    @album = Album.find(params[:id])
+  end
+
+  def new
+    @album = Album.new
+  end
+
+  def edit
+    @album = Album.find(params[:id])
+  end
+
+  def create
+    @album = Album.new(album_params)
+    if @album.save
+      redirect_to album_path(@album.id)
+    else
+      render :new
+    end
+  end
+
+  def update
+    @album = Album.update(params[:id], album_params)
+    if @album.save
+      redirect_to album_path(@album.id)
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+   Album.find(params[:id]).destroy
+   redirect_to albums_path
+  end
+
+  def upvote
+    album = Album.find(params[:id])
```

Comment:
It looks like you search for this album in multiple methods. A possible way you could DRY up your code in a refactor would be to extract finding the album into its own private method and create a `before_action` that calls it before each method `except: [:index, :new, :create]`.